torch/__init__.py

```diff
@@ -1765,6 +1766,199 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
             return cls.ops_table[(op_key, dispatch_key)]
 
 
+################################################################################
+# Exposing torch.export() and related support classes
+################################################################################
+
+# TODO(ycao): Move ExportedProgram, Constraint, dynamic_dim etc to torch.compiler namespace
+
+def export(
+    f: Callable,
+    args: Tuple[Any],
+    kwargs: Optional[Dict[str, Any]] = None,
+    *,
+    constraints: Optional[List["torch._dynamo.eval_frame.Constraint"]] = None,
+) -> "torch._export.exported_program.ExportedProgram":  # type: ignore[name-defined]
+    """
+    torch.export() is a one-shot process for capturing a computation graph from
+    a PyTorch program Ahead-of-Time (AOT).
+
+    This function traces a callable (an nn.Module, a function or a method)
+    containing PyTorch operations and produces an ExportedProgram. The
+    ExportedProgram includes PyTorch operations that perform computations
+    equivalent to those in the given nn.Module or callable.
+
+    In specific terms, torch.export() traces a function `f` by executing it
+    with the provided `args` and `kwargs`. It records the PyTorch operations
+    invoked during execution to produce the ExportedProgram.
+
+
+    **Acceptable input/output types**
+
+    Acceptable types of inputs (for `args` and `kwargs`) and outputs include:
+
+    - Primitive types, i.e. torch.Tensor, int, float, bool, str.
+    - (Nested) Data structures comprising of dict, list, tuple, namedtuple and OrderedDict
+    - Dataclass subclass (must be registered with
```

Comment:
Just say a dataclass? Also, can't we nest a dataclass, i.e., a list of dataclass instances is also acceptable, no?

Reply:
Changed.

we can nest a dataclass